promkit/src/core/json/node.rs

```diff
@@ -330,6 +330,57 @@ impl JsonNode {
         dfs(self, Vec::new(), &mut ret, true, 0); // Start with the root node being visible and is_last true, and indent 0
         ret
     }
+
+
+     pub fn pretty_print(&self, indent: usize) -> String {
+        match self {
+            JsonNode::Object {
+                children,
+                children_visible,
+            } => {
+                let mut output = String::new();
+                let indent_str = " ".repeat(indent);
+                output.push_str(&format!("{{\n"));
+                if *children_visible {
+                    let mut iter = children.iter();
+                    if let Some((key, value)) = iter.next() {
+                        output.push_str(&format!("{}  \"{}\": {}", indent_str, key, value.pretty_print(indent + 2)));
+                    }
+                    for (key, value) in iter {
+                        output.push(',');
+                        output.push('\n');
+                        output.push_str(&format!("{}  \"{}\": {}", indent_str, key, value.pretty_print(indent + 2)));
+                    }
+                    output.push('\n');
+                }
+                output.push_str(&format!("{}}}", indent_str));
+                output
+            }
+            JsonNode::Array {
+                children,
+                children_visible,
+            } => {
+                let mut output = String::new();
+                let indent_str = " ".repeat(indent);
+                output.push_str("[\n");
+                if *children_visible {
+                    let mut iter = children.iter();
+                    if let Some(child) = iter.next() {
+                        output.push_str(&format!("{}{}", indent_str, child.pretty_print(indent + 2)));
+                    }
+                    for child in iter {
+                        output.push(',');
+                        output.push('\n');
+                        output.push_str(&format!("{}{}", indent_str, child.pretty_print(indent + 2)));
+                    }
+                    output.push('\n');
+                }
+                output.push_str(&format!("{}]", indent_str));
+                output
+            }
+            JsonNode::Leaf(value) => format!("{}", value),
+        }
+    }
```

Comment:
I've already shared this elsewhere, but I'll comment here again.

In the current implementation, there's a process that converts `JsonStream` into `Vec<StyledGraphemes>`.
- https://github.com/ynqa/promkit/blob/v0.4.4/promkit/src/core/json/state.rs#L179-L209

This process can be considered as converting each line into a colorized string. Please prepare a common function for these conversions and use it to create strings for the clipboard as well :)

Reply:
Acknowledged; will do! 👍

Reply:
Hey 👋,

Spent a bit of time on this today; will continue later this week.

Regarding your comment on having a common function for the conversions; could you provide a bit of guidance on the how I should implement it? (just some pseudo code would be great).

My assumption right now is that I would add a re-usable method for converting the stream -> Vec<StyledGraphemes> and then having a function for returning the pane and a method to pretty print (which would convert each line of the styledgraphemes into a line of pretty printed json. Is that a good approach?

Reply:
@anton-suprun Your approach sounds good, and I'll add some of my thoughts on it (please feel free to point out any issues).

The actual usage will be at the following location:
- https://github.com/ynqa/jnv/pull/46/files#diff-7d9538a59e520483fe211bbbea9e8535b7d179e52da5bad8c34c44706e480596R200

This means that at the time of use, you should be able to access `json::State` with `self.json` from jnv.

- https://github.com/ynqa/promkit/blob/v0.4.4/promkit/src/core/json/state.rs#L17-L48

So, let’s add a method to json::State.

> Could you provide a bit of guidance on how I should implement it? (just some pseudo code would be great).
> Stream -> Vec and then having a function for returning the pane and a method to pretty print

I envision adding functions like this (the function names will need to be appropriately changed). create_pane would call styled_json, and jnv would call either json_str or styled_json.

```rust
impl State {
    fn styled_json(&self) -> Vec<StyledGraphemes> {
        self
            .stream
            .flatten_kinds()
            .iter()
            .enumerate()
            .map(|(i, kind)| {
                if i == self.stream.cursor.cross_contents_position() {
                    
                } else {
                    
                }
            })
            .collect()
    }

    pub fn json_str(&self) -> String {
        self.styled_json().into_iter().map(|s| s.to_string()).collect::<Vec<String>>().join("\n")
    }
}
```

Reply:
Thanks for the additional explanation 🙏 ; this is roughly what I was thinking but explained it poorly in this message:
>Could you provide a bit of guidance on how I should implement it? (just some pseudo code would be great). Stream -> Vec and then having a function for returning the pane and a method to pretty print

I should have a bit of time later today so hoping to work on it then :)

Reply:
@anton-suprun No problem! Thank you for your continued cooperation 😄

Reply:
@ynqa Oh no; thank you for the guidance and letting be part of jnv 😄 

I didin't have much time today as I hoped; will finish implementing this something this week 👍

Reply:
@anton-suprun Yah 🤘 But please don't overdo it 😉

Reply:
Updated my changes as per your suggestion; thanks again for the help :)
Let me know your thoughts; appreciate any other input

Sorry again for the delay on this 🙏

Reply:
@anton-suprun This changes is no problem for me. Were you planning on adding tests as well?

Reply:
Hi 👋 

Yes, I think I had originally intended to add unit tests (so left the `work-in-progress` comment but this change is quie small so may be safe to proceed without? 

If you would like me to; I'm happy to add unit tests but if it is not required, then maybe we go ahead. 

Let me know your thoughts :)

Reply:
@anton-suprun It seems that tests are not necessary this time. I will decide whether to add tests based on the continuous development.